From 7761778a0440003af420c3745b1c6b0e02205c5d Mon Sep 17 00:00:00 2001
From: Valentine Barshak <valentine.barshak@cogentembedded.com>
Date: Fri, 6 Nov 2020 21:20:49 +0300
Subject: [PATCH] i2c: rcar_i2c: Fix rcar_i2c_xfer callback transfer direction

Enable read direction only for read transfers
when calling rcar_i2c_set_addr function.

Signed-off-by: Valentine Barshak <valentine.barshak@cogentembedded.com>
---
 drivers/i2c/rcar_i2c.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/i2c/rcar_i2c.c b/drivers/i2c/rcar_i2c.c
index 57e0ddb..b6de695 100644
--- a/drivers/i2c/rcar_i2c.c
+++ b/drivers/i2c/rcar_i2c.c
@@ -208,7 +208,7 @@ static int rcar_i2c_xfer(struct udevice *dev, struct i2c_msg *msg, int nmsgs)
 	int ret;
 
 	for (; nmsgs > 0; nmsgs--, msg++) {
-		ret = rcar_i2c_set_addr(dev, msg->addr, 1);
+		ret = rcar_i2c_set_addr(dev, msg->addr, msg->flags & I2C_M_RD);
 		if (ret)
 			return ret;
 
-- 
2.7.4

